script_runner/tests/test_ctx_wrapper.py

```diff
@@ -215,15 +236,17 @@ def test_get_instance_host_ip(self):
         self.assertEqual(result, '1.1.1.1')
 
     def _test_download_resource(self, script, expected_content):
+        result = None
         try:
             result = self._run(script)
-            self.assertTrue(result.startswith('/tmp'))
+            self.assertTrue(result.startswith(tempfile.gettempdir()))
             self.assertTrue(result.endswith('-resource'))
             with open(result) as f:
                 resulting_content = f.read()
             self.assertEqual(expected_content, resulting_content)
         finally:
-            os.remove(result)
```

Comment:
if the `self._run()` line fails, this will throw an UnboundLocalError for `result`, hiding the original exception